### PR TITLE
Fedora 34

### PR DESCRIPTION
**Pull request by KKoukiou**
Locally image-create works as expected. 

 * [x] image-refresh fedora-34

### CODE REVIEW

File: task/testmap.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
             'ubuntu-stable',
             'fedora-32',
             'fedora-33',
+            'fedora-34',
```
Comment (martinpitt) [inline task/testmap.py:36]: This needs to be a separate PR, and please start with adding it to `_manual`. Like this it would immediately run for all cockpit PRs, but we need to sort out the tests first.

File: images/scripts/fedora-34.bootstrap [hunk 2]
```diff
@@ -17,6 +17,11 @@
...
-# Once F34 is released switch to https://dl.fedoraproject.org/pub/fedora/linux/releases/34/Server/x86_64/os/
-$BASE/virt-install-fedora "$1" x86_64 "https://dl.fedoraproject.org/pub/fedora/linux/development/34/Server/x86_64/os/"
+set -ex
+
+# HACK: move back to download.devel after mirroring gets fixed
+URL=https://download.fedoraproject.org/pub/fedora/linux//development/rawhide/Cloud/x86_64/images/
+IMAGE=$(curl -L -s "$URL" | sed -n '/<a href=.*Fedora-Cloud-Base-Rawhide.*qcow2"/ { s/^.*href="//; s_".*$__; p }')
+[ -n "$IMAGE" ]
+
+exec $(dirname $0)/lib/cloudimage.bootstrap "$1" "$URL/$IMAGE"
```
Comment (martinpitt) [inline images/scripts/fedora-34.bootstrap:27]: Please squash this into the first commit.

File: images/scripts/fedora-34.setup [hunk 3]
```diff
@@ -0,0 +1,193 @@
+#!/bin/bash
```
Comment (martinpitt) [inline images/scripts/fedora-34.setup:1]: This file should be a symlink to fedora.setup, not a copy, like fedora-{32,33}.setup. Let's please avoid divergence, and put f-34 special cases right into the setup.

**Review by martinpitt — CHANGES_REQUESTED** [review]

**Comment by cockpituous** [comment]
image-refresh in progress on 4-ci-srv-03.
Log: https://logs.cockpit-project.org/logs/image-refresh-1685-20210216-104545/

**Comment by cockpituous** [comment]
image-refresh fedora-33 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-33-20210216-110505

**Comment by cockpituous** [comment]
image-refresh in progress on 2-cockpit-10.
Log: https://logs.cockpit-project.org/logs/image-refresh-1685-20210216-110707/

**Comment by cockpituous** [comment]
image-refresh fedora-34 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-34-20210216-110808

**Comment by martinpitt** [comment]
That will not work, you need to add the fedora-34.setup symlink.

**Comment by cockpituous** [comment]
image-refresh in progress on 1-ci-srv-02.
Log: https://logs.cockpit-project.org/logs/image-refresh-1685-20210216-111313/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-1685-20210216-111313/

**Comment by cockpituous** [comment]
image-refresh in progress on 4-cockpit-7.
Log: https://logs.cockpit-project.org/logs/image-refresh-1685-20210216-113030/

**Comment by cockpituous** [comment]
image-refresh fedora-34 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-34-20210216-115151

File: images/scripts/fedora-34.bootstrap [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301 USA.
+
+set -ex
+
+# HACK: move back to download.devel after mirroring gets fixed
```
Comment (martinpitt) [inline images/scripts/fedora-34.bootstrap:22]: This is a leftover from the rhel bootstrap file, and just confusing here.

File: images/scripts/fedora.setup [hunk 5]
```diff
@@ -105,6 +107,12 @@ libreport-plugin-bugzilla \
 libreport-fedora
 "
 
+if [ "$IMAGE" = "fedora-34" ]; then
+    # npm was replaced with ntpsec
```
Comment (martinpitt) [inline images/scripts/fedora.setup:111]: not npm :-)

File: images/scripts/fedora.setup [hunk 6]
```diff
@@ -114,7 +122,18 @@ maybe firewall-cmd --permanent --add-port 8765/tcp
...
+# These build from a cloud image, set up network-scripts for eth1 to avoid long boot hang on
+# NetworkManager-wait-online.service
+if [ "$IMAGE" = "fedora-34" ]; then
+    printf 'BOOTPROTO=none\nDEVICE=eth1\nONBOOT=no\n' > /etc/sysconfig/network-scripts/ifcfg-eth1
+    # our cloud-init.iso does not set up the host name
+    echo "127.0.1.1 $(hostname)" >> /etc/hosts
+fi
+
+# admin user exists in the cloud image
+if [ "$IMAGE" != fedora-34 ]; then
```
Comment (martinpitt) [inline images/scripts/fedora.setup:134]: Right, we specifically told it so through our cloud-init.iso. Could you please use the same approach as in rhel.setup?
```sh
# cloud images already create the user through cloud-init.iso
if ! getent passwd admin >/dev/null; then
    useradd -c Administrator -G wheel admin
    echo foobar | passwd --stdin admin
fi
```

**Review by martinpitt — CHANGES_REQUESTED** [review]
Nice, thank you! A few minor issues, but by and large this looks good. I'm glad that we move to an "official" product from our totally custom previous images.

File: images/scripts/fedora.setup (hunk 5 above)
Comment (KKoukiou) [reply images/scripts/fedora.setup:111]: haah too many times written npm

**Comment by cockpituous** [comment]
image-refresh in progress on 2-cockpit-8.
Log: https://logs.cockpit-project.org/logs/image-refresh-1685-20210224-090707/

**Comment by cockpituous** [comment]
image-refresh fedora-34 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-34-20210224-092020

**Review by martinpitt — APPROVED** [review]
The image generally works, although there are still [25 failures](https://github.com/cockpit-project/cockpit/pull/15350) -- but I propose we handle them piece by piece, instead of blocking this PR forever and letting the image time out and this PR get out of sync. Thank you!

**Comment by martinpitt** [comment]
I'd also like to land this now so that we can start testing starter-kit and cockpit-podman on F34.
